tests/back/Pim/Enrichment/Specification/Component/Product/Model/ProductModelSpec.php

```diff
@@ -717,6 +720,358 @@ function it_is_updated_when_adding_a_non_empty_association(
+        $xsellType = new AssociationType();
+        $xsellType->setCode('x_sell');
+        $xsellAssociation->setAssociationType($xsellType);
+        $xsellAssociation->addProduct($product);
+
+        $this->addAssociation($xsellAssociation);
+        $this->cleanup();
+
+        $this->addAssociatedProduct($product, 'x_sell');
+        $this->isDirty()->shouldBe(false);
+    }
+
+    public function it_removes_a_product_from_an_association(
+        AssociationInterface $association
+    ): void {
+        $product = new Product();
+        $xsellType = new AssociationType();
+        $xsellType->setCode('x_sell');
+        $association->getAssociationType()->willReturn($xsellType);
+        $association->hasProduct($product)->willReturn(true);
+        $association->getProducts()->willReturn(new ArrayCollection([$product]));
+        $association->getProductModels()->willReturn(new ArrayCollection([]));
+        $association->getGroups()->willReturn(new ArrayCollection([]));
+        $association->setOwner($this)->willReturn($association);
+        $this->addAssociation($association);
+
+        $association->removeProduct($product)->shouldBeCalled();
+
+        $this->removeAssociatedProduct($product, 'x_sell');
+    }
+
+    public function it_is_updated_if_a_product_is_removed_from_an_association(): void {
+        $product = new Product();
+        $xsellAssociation = new ProductAssociation();
+        $xsellType = new AssociationType();
+        $xsellType->setCode('x_sell');
+        $xsellAssociation->setAssociationType($xsellType);
+        $xsellAssociation->addProduct($product);
+
+        $this->addAssociation($xsellAssociation);
+        $this->cleanup();
+
+        $this->removeAssociatedProduct($product, 'x_sell');
+        $this->isDirty()->shouldBe(true);
+    }
+
+    public function it_is_not_updated_if_a_product_to_remove_from_an_association_does_not_exist(): void {
+        $product = new Product();
+        $xsellAssociation = new ProductAssociation();
+        $xsellType = new AssociationType();
+        $xsellType->setCode('x_sell');
+        $xsellAssociation->setAssociationType($xsellType);
+
+        $this->addAssociation($xsellAssociation);
+        $this->cleanup();
+
+        $this->removeAssociatedProduct($product, 'x_sell');
+        $this->isDirty()->shouldBe(false);
+    }
+
+    public function it_returns_associated_products_in_terms_of_an_association_type(): void
+    {
+        $plate = new Product();
+        $spoon = new Product();
+
+        $xsellAssociation = new ProductAssociation();
+        $xsellType = new AssociationType();
+        $xsellType->setCode('x_sell');
+        $xsellAssociation->setAssociationType($xsellType);
+        $xsellAssociation->addProduct($plate);
+        $xsellAssociation->addProduct($spoon);
+
+        $this->addAssociation($xsellAssociation);
+
+        $this->getAssociatedProducts('x_sell')->shouldBeLike(new ArrayCollection([$plate, $spoon]));
```

Comment:
Same for getAssociatedProductModels and getAssociatedGroups and in ProductSpec

```suggestion
        $this->getAssociatedProducts('x_sell')->shouldBeLike(new ArrayCollection([$plate, $spoon]));
        $this->getAssociatedProducts('another_association_type_code')->shouldBeLike(new ArrayCollection([$plate, $spoon])); 
```